mantaray/received.py

```diff
@@ -316,14 +332,31 @@ def _handle_cap(server_view: views.ServerView, args: list[str]) -> None:
     subcommand = args[1]
     if subcommand == "ACK":
         acknowledged = set(args[-1].split())
+
         if "sasl" in acknowledged:
             server_view.core.send("AUTHENTICATE PLAIN")
+
+        for capability in acknowledged:
+            server_view.core.cap_list.add(capability)
+
     elif subcommand == "NAK":
         rejected = set(args[-1].split())
         if "sasl" in rejected:
             # TODO: this good?
             raise ValueError("The server does not support SASL.")
 
+    server_view.core.pending_cap_count -= 1
```

Comment:
```suggestion
    server_view.core.pending_cap_count -= len(rejected)
```

Or if you are sure that it cannot NAK multiple capabilities at once, change the code above instead.

Reply:
@Akuli It can NAK multiple capabilities at once if you do `CAP REQ capability_1 capability_2 capability_3`.

Also, if we get an `ACK`, `pending_cap_count` gets `-= 0`, then it will never send `CAP END`. It has to do at least `-= 1`

Reply:
So it should treat `args[-1]` as a single capability, instead of splitting it?

Reply:
Another thing I just noticed: The `if`,`elif` has no `else`, but it should probably have one. If the server sends `CAP FOO`, it should be either ignored silently, or it should be an error. Currently that would decrement `pending_cap_count` and potentially cause weird problems.